packages/remix-forms/src/createForm.tsx

```diff
@@ -422,8 +422,7 @@ function createForm({
           : transition.state === 'submitting'
 
       setDisabled(shouldDisable)
-      // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [transition.state, formState])
+    }, [transition.state, formState, mode, isValid])
```

Comment:
I think this addition makes sense as one value is a string union and the other is a boolean, if they change this code should probably re-run... any ideas?